lib/util/LocationStatusManager.js

```diff
@@ -12,20 +12,156 @@ const actions = {
+        this.service = service;
+        this.location = location;
+        this.state = initialState;
+        this.scheduledResumeDate = null;
+        this.pendingOperations = new Map();
+    }
+
+    canExecuteOperation(operationType) {
+        const validOperations = {
+            [SERVICE_STATES.RUNNING]: [OPERATION_TYPES.PAUSE],
+            [SERVICE_STATES.PAUSED]: [OPERATION_TYPES.RESUME, OPERATION_TYPES.SCHEDULE_RESUME],
+            [SERVICE_STATES.SCHEDULED_RESUME]: [OPERATION_TYPES.SCHEDULE_RESUME,
+                OPERATION_TYPES.RESUME, OPERATION_TYPES.DELETE_SCHEDULE],
+        };
+        return validOperations[this.state].includes(operationType);
+    }
+
+    getConflictingOperations(newOperation) {
+        const conflicts = [];
+        for (const [, operation] of this.pendingOperations) {
+            if (operation.state === OPERATION_STATES.PROCESSING) {
+                if (this.areOperationsConflicting(operation.type, newOperation.type)) {
+                    conflicts.push(operation);
+                }
+            }
+        }
+        return conflicts;
+    }
+
+    areOperationsConflicting(operation1, operation2) {
+        const conflictMatrix = {
+            [OPERATION_TYPES.PAUSE]: [OPERATION_TYPES.RESUME, OPERATION_TYPES.SCHEDULE_RESUME],
+            [OPERATION_TYPES.RESUME]: [OPERATION_TYPES.PAUSE, OPERATION_TYPES.SCHEDULE_RESUME],
+            [OPERATION_TYPES.SCHEDULE_RESUME]: [OPERATION_TYPES.PAUSE, OPERATION_TYPES.RESUME, 
+                OPERATION_TYPES.DELETE_SCHEDULE],
+            [OPERATION_TYPES.DELETE_SCHEDULE]: [OPERATION_TYPES.SCHEDULE_RESUME],
+        };
+        return conflictMatrix[operation1]?.includes(operation2) || conflictMatrix[operation2]?.includes(operation1);
+    }
+
+    addPendingOperation(operation) {
+        return this.pendingOperations.set(operation.id, operation);
+    }
+
+    removePendingOperation(operationId) {
+        return this.pendingOperations.delete(operationId);
+    }
+
+    executeStateTransition(operationType, options = {}) {
+        const transitions = {
+            [OPERATION_TYPES.PAUSE]: () => {
+                this.state = SERVICE_STATES.PAUSED;
+                this.scheduledResumeDate = null;
+            },
+            [OPERATION_TYPES.RESUME]: () => {
+                this.state = SERVICE_STATES.RUNNING;
+                this.scheduledResumeDate = null;
+            },
+            [OPERATION_TYPES.SCHEDULE_RESUME]: () => {
+                this.state = SERVICE_STATES.SCHEDULED_RESUME;
+                this.scheduledResumeDate = options.scheduleDate;
+            },
+            [OPERATION_TYPES.DELETE_SCHEDULE]: () => {
+                this.state = SERVICE_STATES.PAUSED;
+                this.scheduledResumeDate = null;
+            },
+        };
+
+        const transition = transitions[operationType];
+        if (transition) {
+            transition();
+        }
```

Comment:
```suggestion
            transition?.();
```

+ A log in case there is no transition?